### DISCUSSION TITLE
Issue getting the number of instances

### DISCUSSION THREAD

**Discussion by FengZh61** [ccfos/nightingale#1241] [Q&A]
![image](https://user-images.githubusercontent.com/98309003/198958219-56310032-b0db-4dc4-be79-8598fd35e9fa.png)

我在这里过滤ident标签，共过滤出4个，但是实际我系统中只有2个，另外2个是之前的，在prometheus中和vm中都查不到了，在快捷视图为啥还能查到？

![image](https://user-images.githubusercontent.com/98309003/198958734-fe375733-e818-4fb9-abc0-93b59376519d.png)

**Comment by UlricQin** [comment]
这个ident列表是调用的这个接口获取的：`/api/n9e/prometheus/api/v1/label/ident/values`参数是：
```
start: 
1667284576
end: 
1667288176
match[]: 
{__name__="mem_available_percent"}
```

你打开chrome开发者工具看一下就知道了。这个接口是n9e-webapi的接口，但是n9e-webapi啥都没做，直接把这个请求proxy给后端时序库了，后端时序库把之前的老数据返回了。

你调用 prometheus 的 `/api/v1/label/ident/values` 也会是一样的效果。

如果不想看到老旧数据，可以google一下“prometheus delete_series” 使用删除接口把没用的数据删除

**Reply by FengZh61** [reply]
> 这个ident列表是调用的这个接口获取的：`/api/n9e/prometheus/api/v1/label/ident/values`参数是：
> 
> ```
> start: 
> 1667284576
> end: 
> 1667288176
> match[]: 
> {__name__="mem_available_percent"}
> ```
> 
> 你打开chrome开发者工具看一下就知道了。这个接口是n9e-webapi的接口，但是n9e-webapi啥都没做，直接把这个请求proxy给后端时序库了，后端时序库把之前的老数据返回了。
> 
> 你调用 prometheus 的 `/api/v1/label/ident/values` 也会是一样的效果。
> 
> 如果不想看到老旧数据，可以google一下“prometheus delete_series” 使用删除接口把没用的数据删除

我理解您说的那个意思，我看过n9e接口的参数，查询时间是当前时间一小时内的数据，但是1小时内的数据已经不存在多余的2个ident了，要查3个小时或者更久的的时间才能查到。这种情况是否正常呢？
我今天早上再看，ident就只剩下2个了。我查2天前的数据是能查到旧的ident。
